### DISCUSSION TITLE
Is it possible to import url dynamic?

### DISCUSSION THREAD

**Discussion by EloB** [vercel/next.js#33620] [Help]
I'm not sure if this is possible or bad but would be kind of cool if I can implement these to together somehow.
https://nextjs.org/docs/api-reference/next.config.js/url-imports
https://nextjs.org/docs/advanced-features/dynamic-import
https://www.framer.com/docs/guides/handshake

So I can add a way to include "snippet" of components into a CMS. So marketing could go crazy on the marketing website.

Another thought might be to make some type of integration between CMS and the deploy system so it triggers a build but then installing "new" snippet in CMS would take awhile.

Is there possible to have some type of runtime url import?

**Comment by EloB** [comment]
I made a codesandbox that doesn't work but shows the idea.
https://codesandbox.io/s/frosty-butterfly-p9zq0?file=/pages/index.tsx

**Comment by JuHwon** [comment]
I was also just searching for this feature. Would love to load SVGs from the CMS.

**Comment by multiplehats** [comment]
Did any of you figure this out @JuHwon / @EloB ? I'm also trying to get this to work

**Reply by multiplehats** [reply]
I did try to add this before the import, that gave me a different error @EloB 

```
 [url]: (await import(/* webpackIgnore: true */ url)).default as ComponentType<any>,
```

**Comment by multiplehats** [comment]
Follow up, pretty sure this is not possible: https://github.com/vercel/next.js/issues/36895

**Reply by trhinehart-godaddy** [reply]
Just ran into this as well, the desired feature is to import a url at runtime, not build time.

**Reply by zain-bin-ramzan** [reply]
It seems that it is not possible. I have faced something similar while working with the Matterport SDK
```
async fetch(e, t) {
  try {
    const s = await import(e), 
      r = Object.assign({}, s);
    if (t && t in window) {
      const e = window[t] || {};
      Object.assign(r, e), delete window[t];
    }
    this.libraryResolver(r);
  } catch (e) {
    this.libraryRejecter(e);
  }
}
```
